tests/framework/prototype.py

```diff
@@ -0,0 +1,279 @@
+import yaml
+from typing import Optional
+from kubernetes import client, config, utils
+import os
+import time
+import docker
+import logging
+import unittest
+import jsonpatch
+
+# Utility functions
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+def search_path(cr, steps):
```

Comment:
Could you add a doc string for this function?

Reply:
Added.